2-copy-of-code/lesson-14/data/cart.js

```diff
@@ -57,4 +57,18 @@ export function calculateCartQuantity() {
   });
 
   return cartQuantity;
+}
```

Comment:
could someone explain to me why this works? i am at a total loss here. i would imagine that we delete the original cartItem with splice() and then push the new item with a new quantity into the cart. but here we don't do any of that. we don't delete the previous object nor do we push anything into the cart. yet it works perfectly and i can't understand why that is. thank you to anyone who can be so kind as to help me understand this.

Reply:
@thatideadude 
When we call the updateQuantity function, we pass in the productId and newQuantity. The variable matchingItem is initially declared as undefined. The cart contains the products, so we loop through the cart and check if the passed productId matches any productId in the cart. If a match is found, that product is assigned to matchingItem.

We then update the quantity property of the matchingItem to the newQuantity. Since the cart is an array of objects, and each product is stored as an object, updating the quantity property directly modifies the product in the cart.

Reply:
Thanks so much for taking the time to help Mohan! This definitely works, i still don't understand why though. Is it not possible to have two objects with the same productId inside the cart? Is that what's causing the matchingitem.quantity to write over the original cartitem.quantity?

As i was trying to wrap my head around it last week, i found a work around that cuts a few lines of code. It basically just skips the matchingItem altogether.

cart.forEach((item) => {
    if (productId === item.productId) {
      item.quantity = newQuantity;

This is easier for me to understand, but I have a feeling that the purpose of this exercise was to teach me something greater that i failed to grasp.